### ISSUE TITLE
Some illegal annotation expression forms are not flagged as errors

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6800]
Pyright doesn't currently generate an error for incorrect use of unary, binary, or ternary operators in type annotations.

```python
v1: int if 1 < 2 else str
v2: 1 + 2
v3: -2
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.343, which I just published. It will also be included in a future release of pylance.

**Comment by hmc-cs-mdrissi** [comment]
I'm guessing it's related to this change, but I have some type aliases defined like,

`GraphFeatureTowerLayer = Layer[tf.Tensor, tf.Tensor] if TYPE_CHECKING else Layer`

that now give an error message `Ternary expression not allowed in type annotation`. The alias needs `TYPE_CHECKING` as class is generic in stubs but not at runtime (also common in standard library).

I can workaround it by expanding to,

```
if TYPE_CHECKING:
  GraphFeatureTowerLayer = ...
else:
  GraphFeatureTowerLayer = ...
```

**Comment by erictraut** [comment]
@hmc-cs-mdrissi, this is intended. As the error message says, ternary expressions are not permitted within type expressions. The proper way to do this is as you've specified in your workaround.
